docs/07-templated-components.md

```diff
@@ -1 +1,246 @@
+    <IsPackable>true</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- .js/.css files will be referenced via <script>/<link> tags; other content files will just be included in the app's 'dist' directory without any tags referencing them -->
+    <EmbeddedResource Include="content\**\*.js" LogicalName="blazor:js:%(RecursiveDir)%(Filename)%(Extension)" />
+    <EmbeddedResource Include="content\**\*.css" LogicalName="blazor:css:%(RecursiveDir)%(Filename)%(Extension)" />
+    <EmbeddedResource Include="content\**" Exclude="**\*.js;**\*.css" LogicalName="blazor:file:%(RecursiveDir)%(Filename)%(Extension)" />
+  </ItemGroup>  
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Blazor.Browser" Version="0.7.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Blazor.Build" Version="0.7.0" PrivateAssets="all" />
+  </ItemGroup>
+
+</Project>
+```
+
+There are a few things here worth understanding. 
+
+Firstly, it's recommended that all Blazor project targets C# version 7.3 or newer (`<LangVersion>7.3</LangVersion>`), Blazor relies on some features added in version 7.3 to make event handlers work correctly.
+
+Next, the `<IsPackable>true</IsPackable>` line makes it possible the create a NuGet package from this project. We won't be using this project as a package in this example, but this is a good thing to have for a class library.
+
+Next, the lines that look like `<EmbeddedResource ... />` give the class library project special handling of content files that should be included in the project. This makes it easier to do multi-project development with static assests, and to redistibute libraries containing static assets. We'll see this in action later.
+
+Lastly the two `<PackageReference />` elements add package references to the Blazor runtime libraries and the Blazor build support.
+
+At this point the library is a now a fully-fledged *Blazor class library*, in the following steps we will add some components.
+
+## Writing a templated dialog
+
+We are going to revisit the dialog system that is part of `Index` and turn it into something that's decoupled from the application.
+
+Let's think about how a *reusable dialog* should work. We would expect a dialog component to handle showing and hiding itself, as well as maybe styling to appear visually as a dialog. However, to be truly reusable, we need to be able to provide the content for the inside of the dialog. We call a component that accepts *content* as a parameter a *templated component*.
+
+Blazor happens to have a feature that works for exactly this case, it's similar to how a layout works. Recall that a layout has a `Body` parameter, and the layout gets to place other content *around* the `Body`. In a layout, the `Body` parameter is of type `RenderFragment` which is a delegate type that the runtim has special handling for. The good news is that this feature is not limited layouts. Any component can declare a parameter of type `RenderFragment`.
```

Comment:
This is a great explanation. I like the comparison with layout.

Reply:
`runtim` -> `runtime` (or perhaps, `framework`).

`not limited layouts` -> `not limited to layouts`

Reply:
Thanks, this is what happens when I'm not typing on a tiny table with a loud jazz band 😁  However there is a typo here: `runtim` -> `runtime`.

Reply:
Also `not limited layouts` -> `not limited to layouts`